### PR TITLE
Add Search input

### PR DESCRIPTION
**Pull request by JaneEllison**
Add Search input, - with styles, liveSearching.
No Redux at this time.

### CODE REVIEW

File: src/components/SearchInput/SearchInput.jsx [hunk 1]
```diff
@@ -1,7 +1,62 @@
...
+import { dataService } from '../../services/dataService';
 
 function SearchInput() {
-  return <div>Search input</div>;
+  const [searchQuery, setSearchQuery] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
+
+  useEffect(() => {
+    if (searchQuery) {
+      searchCities();
```
Comment (anton-vlasik) [inline src/components/SearchInput/SearchInput.jsx]: again I would suggest to pass search query as a param here. What if you want to move searchCities out of this component's scope

File: src/components/SearchInput/SearchInput.jsx [hunk 2]
```diff
@@ -1,7 +1,62 @@
...
+    }
+  }, [searchQuery]);
```
Comment (anton-vlasik) [inline src/components/SearchInput/SearchInput.jsx]: Also as an improvement - think how can you optimize your code. Right now on every key stroke you are sending a request. Could you **debounce** it?

Comment (JaneEllison) [reply src/components/SearchInput/SearchInput.jsx]: Done

File: src/components/SearchInput/SearchInput.jsx (hunk 1 above)
Comment (JaneEllison) [reply src/components/SearchInput/SearchInput.jsx]: It's true, corrected

File: src/components/SearchInput/SearchInput.jsx [hunk 3]
```diff
@@ -1,7 +1,63 @@
...
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  const handleChange = e => {
+    setSearchQuery(e.target.value);
+  };
+
+  const renderSearchResults = searchResults.map(item => (
```
Comment (anton-vlasik) [inline src/components/SearchInput/SearchInput.jsx]: I don't think there is a lot of added value to move this mapping here. It is super usual to write it "inline" right n JSX

File: src/components/SearchInput/SearchInput.jsx [hunk 4]
```diff
@@ -1,7 +1,63 @@
...
+    <li key={item.id} className={styles.searchResultItem}>
+      <img src={SearchLocationIcon} alt="location" />
+      <span>{item.name}</span>
+    </li>
+  ));
+
+  return (
+    <div className={styles.searchWrapper}>
+      <div className={styles.searchForm}>
+        <DebounceInput
```
Comment (anton-vlasik) [inline src/components/SearchInput/SearchInput.jsx]: I want you to write it yourself, please. I allow you to use debounce function from lodash.debounce. But Debounce input should be written by yourself. Don't forget to remove unneeded packages after that

File: src/components/SearchInput/SearchInput.jsx (hunk 3 above)
Comment (JaneEllison) [reply src/components/SearchInput/SearchInput.jsx]: ok, got it

File: src/components/SearchInput/SearchInput.jsx (hunk 4 above)
Comment (JaneEllison) [reply src/components/SearchInput/SearchInput.jsx]: i did it

File: src/hooks/useDebounce.jsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { useState, useEffect } from 'react';
+
+export function useDebounce(value, delay) {
+  const [dbValue, setDbValue] = useState('');
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
```
Comment (anton-vlasik) [inline src/hooks/useDebounce.jsx]: This is more of a timeout Id.

File: src/components/SearchInput/SearchInput.jsx [hunk 6]
```diff
@@ -1,21 +1,22 @@
...
 import SearchIcon from '../../assets/images/search-icon.png';
 import SearchLocationIcon from '../../assets/images/search-location-icon.png';
-import { NUMBER_OF_RESULTS } from '../../constants/forecaApi';
-import { dataService } from '../../services/dataService';
 
 function SearchInput() {
   const [searchQuery, setSearchQuery] = useState('');
   const [searchResults, setSearchResults] = useState([]);
 
+  const debouncedSearchQuery = useCallback(useDebounce(searchQuery, 1000));
```
Comment (anton-vlasik) [inline src/components/SearchInput/SearchInput.jsx]: useCallback accepts a function... and a list of dependencies, if you don't want it to run every time returning you new value. In your case you are passing a string to it from what I see. So I'm a bit confused.

File: src/hooks/useDebounce.jsx (hunk 5 above)
Comment (JaneEllison) [reply src/hooks/useDebounce.jsx]: Ok, fixed

File: src/components/SearchInput/SearchInput.jsx (hunk 6 above)
Comment (JaneEllison) [reply src/components/SearchInput/SearchInput.jsx]: Yes. I rewrote function and forget to remove this one. Fixed.

**Review by anton-vlasik — APPROVED** [review]

**Comment by ximet** [comment]
resolve conflict

**Review by ximet — APPROVED** [review]

**Pull request merged**